openapi_python_client/templates/property_templates/union_property.pyi

```diff
@@ -39,9 +39,9 @@ elif {{ source }} is None:
 {% endif %}
 {% for inner_property in property.inner_properties %}
     {% if loop.first and property.required and not property.nullable %}{# No if UNSET or if None statement before this #}
-if isinstance({{ source }}, {{ inner_property.get_type_string(no_optional=True) }}):
+if isinstance({{ source }}, {{ inner_property.get_instance_type_string() }}):
     {% elif not loop.last %}
-elif isinstance({{ source }}, {{ inner_property.get_type_string(no_optional=True) }}):
+elif isinstance({{ source }}, {{ inner_property.get_instance_type_string() }}):
     {% else %}
 else:
     {% endif %}
```

Comment:
Similar story here I think, we just change the order of some checks. Something like 
```
{% for inner_property in (inner_property for inner_property in property.inner_properties if inner_property.template) %}
{% if loop.first and property.required and not property.nullable %}{# No if UNSET or if None statement before this #}
if isinstance({{ source }}, {{ inner_property.get_instance_type_string() }}):
{% else %}
elif isinstance({{ source }}, {{ inner_property.get_instance_type_string() }}):
{% from "property_templates/" + inner_property.template import transform %}
    {{ transform(inner_property, source, destination, declare_type=False) | indent(4) }}
{% endif %}
{% endfor %}
else:
    {{ destination }} = {{ source }}
```
Editing in GH UI, so probably some mistakes made, but hopefully you get the idea.